spyder/plugins/variableexplorer/widgets/arrayeditor.py

```diff
@@ -1131,6 +1131,8 @@ def show_preferences_dialog(self) -> None:
         # Create dialog using current options
         dialog = PreferencesDialog('array', parent=self)
         dialog.float_format = self.arraywidget.model.get_format_spec()
+        if dialog.float_format == 'd':
+            dialog.float_format = '.0f'
```

Comment:
This is not the right place to do this because we haven't processed yet the dialog info introduced by the user. Instead, please move these lines below line **1143**, like this:

```python
if float_format == 'd':
    float_format = '.0f'
```

In addition, please do the same for our dataframe editor, specifically here to have feature parity between the two editors:

https://github.com/spyder-ide/spyder/blob/ce927ac4703ad4872fe70122998252ba759f9462/spyder/plugins/variableexplorer/widgets/dataframeeditor.py#L2275-L2276